### ISSUE TITLE
In-app descriptor editor mode (built on --simulate-all)

### ISSUE THREAD

**Issue by mmaher88** [mmaher88/logitune#33]
## Why

Logitune ships JSON device descriptors that hand-curate the layout of buttons, scroll wheels, and easy-switch indicators on each Logitech mouse. Today these descriptors are produced by the Options+ extractor, then iteratively patched by hand because the source data has gaps (MX Vertical has no back image; LIFT uses split metadata files; m720-triathlon has channel buttons on top, not the back; etc.).

The current iteration loop is:
1. Open the JSON in a text editor
2. Edit `xPct`/`yPct` by guess
3. Restart Logitune
4. Squint at whether the slot circle landed in the right place
5. Repeat

This is slow, error-prone, and impossible to do for 30+ community devices in any reasonable time. We need an in-app editor that lets us drag elements directly on the rendered device pages and save back to the source descriptor file.

## Two-part scope

This issue covers two pieces that ship together on branch `editor-mode`:

### Part 1 — `--simulate-all` debug flag (foundation, already implemented)

A CLI flag that populates the carousel with one fake card per descriptor in `DeviceRegistry` instead of scanning udev for real hardware. Lets the maintainer walk every community descriptor in a single window without owning the physical mice.

```
$ logitune --simulate-all
```

| Layer | Production | --simulate-all |
|---|---|---|
| udev / hidraw / HID++ probing | yes | no |
| Capability dispatch tables | yes | no |
| Battery poll, button divert, keystroke injection | yes | no |
| DeviceRegistry JSON load | yes | yes |
| Descriptor → DeviceModel pipeline | yes | yes |
| QML carousel + device pages render | yes | yes |
| Hotspot positioning, `kind` lookup, `isThumbWheel` | yes | yes |

Already implemented in three files (~125 lines, all guarded by the flag): `main.cpp` adds `QCommandLineParser`, `AppController::startMonitoring(simulateAll)` branches between real and simulated, `DeviceManager::simulateAllFromRegistry()` synthesizes a `PhysicalDevice` + `DeviceSession` per descriptor, `DeviceSession::applySimulation()` stubs the connected state.

### Part 2 — In-app descriptor editor (new, design approved)

`--simulate-all` lets you *see* what's wrong; this lets you *fix* it without leaving the app. Adds an `--edit` flag (orthogonal to `--simulate-all`) that gates a WYSIWYG editor: drag easy-switch slot circles, button hotspot markers, and overlay cards directly on the existing pages, with changes saved back to the descriptor JSON the device was loaded from.

**Editing surface (4 tiers):**

1. **Easy-switch slots** — drag the slot circles on the back image to set `xPct`/`yPct`.
2. **Button & point/scroll hotspots** — drag markers (anchor on the device image) and overlay cards (`side` + `labelOffsetYPct`) independently. Connector line auto-tracks.
3. **Image upload** — drag-and-drop a PNG onto the image area, or use a "Replace image" button that opens a file picker. Atomic copy into the descriptor's directory.
4. **Schema additions + text editing** — two new optional fields (`label` on slot objects, `displayName` on hotspots) plus in-place double-click editing of slot labels, button names, and the device-name header.

**Out of scope (deliberately):**
- Editing protocol-level fields (`controlId`, `buttonIndex`, DPI ranges, `capabilities` flags) — these come from the hardware
- Carousel ordering, custom themes, custom icons
- Multi-window coordination
- "Publish to community-devices repo" button (deferred)
- Auto-save / drafts (manual save only)

**Key UX decisions:**
- WYSIWYG (in-place on existing pages, not a separate editor view) — fine-tuning `labelOffsetYPct` matters because cards overlap text in the *real* page, not in an abstract editor canvas
- Manual save with "unsaved changes" indicator — drag updates are in-memory until explicit save
- Per-device undo/redo, persistent across device switches until save or quit
- File watcher detects external changes (`git pull` on community-devices repo); silent reload when no pending edits, conflict prompt with diff viewer when there are
- Atomic save via `QSaveFile`; preserves unknown fields in the source file (writer applies edits to parsed-from-disk JSON, never re-derives from `JsonDevice`)
- Zero impact on production binary path: editor code only runs when `--edit` is set

## Architecture

The design follows the existing MVVM split. New pieces:

- **`EditorModel`** (ViewModel, QML singleton, instantiated only when `--edit` is set) — owns per-device pending edits, undo stacks, dirty tracking, file watcher, save flow
- **`DescriptorWriter`** (Model layer, pure C++) — atomic round-trip JSON writer with preserve-unknown-fields
- **`JsonDevice::sourcePath()`** — adds source-path tracking so the editor knows where to write back
- **`DeviceRegistry::reload(path)`** — single entry point used by both save and the file watcher
- **`EditorToolbar.qml`** — shared 36px bar (save / reset / undo / redo / unsaved indicator), visible only in edit mode
- **Drag handlers** added inline to `EasySwitchPage.qml`, `ButtonsPage.qml`, `PointScrollPage.qml`, gated on `EditorModel.editing`

Full design at `docs/superpowers/specs/2026-04-15-editor-mode-design.md` (committed on `editor-mode` branch).

## Build sequence

1. Foundation: `JsonDevice::sourcePath()`, `DeviceRegistry::loadFromFile`/`reload`, schema additions
2. `DescriptorWriter` with round-trip preservation
3. `EditorModel` (no QML yet) — state, undo, save, file watcher
4. CLI `--edit` flag + `AppController` plumbing
5. `EditorToolbar.qml`
6. Tier 1 — easy-switch slot drag (end-to-end first slice; fixes the original use case)
7. Tier 2 — button & point/scroll markers, cards, text editing
8. Tier 3 — image upload (drop area + file picker)
9. Conflict banner UI for external file changes (with diff modal)
10. Manual smoke-test pass against a copy of the community-devices repo

By step 6 the maintainer can already fix easy-switch positions in-app — exactly the workflow that triggered this whole evolution.

## Use case

```
# Bulk curation: walk and edit every community descriptor
git clone https://github.com/mmaher88/logitune-devices.git /tmp/logitune-devices
mkdir -p /tmp/sim/logitune
ln -s /tmp/logitune-devices /tmp/sim/logitune/devices
XDG_DATA_HOME=/tmp/sim logitune --simulate-all --edit

# Or: edit the descriptor of the real plugged-in device
logitune --edit
```

## Out of scope (tracked but deferred)

- Real HID++ command replay or button-press simulation
- Per-device simulation (`--simulate-device <slug>`)
- "Publish to community-devices repo" button
